store/views.py

```diff
@@ -4,79 +4,122 @@
 from django.contrib.auth.decorators import login_required
 from django.http import JsonResponse
 from django.views.decorators.csrf import csrf_exempt
-
-# Create your views here.
-
+from datetime import date
 def index(request):
     return render(request, 'store/index.html')
 
 def bookDetailView(request, bid):
     template_name = 'store/book_detail.html'
+    book = Book.objects.get(id=bid)
+    availbooks= BookCopy.objects.filter(book=book,status=True)
+    i=availbooks.count()
     context = {
-        'book': None, # set this to an instance of the required book
-        'num_available': None, # set this to the number of copies of the book available, or 0 if the book isn't available
+        'book': book,
+        'num_available': i,
     }
-    # START YOUR CODE HERE
-    
-    
     return render(request, template_name, context=context)
 
 
 @csrf_exempt
 def bookListView(request):
     template_name = 'store/book_list.html'
+    # Searching 
+    bookdata = Book.objects.all()
+    get_data = request.GET
+    some= get_data.keys()
+    if (len(some)!=0):
+        if get_data['title'] =='' :
+            if get_data['author']=='' :
+                if get_data['genre']=='' :
+                    bookdata = Book.objects.all()
+                else: 
+                    bookdata = Book.objects.filter(genre=get_data['genre'])
+            else:
+                if get_data['genre']=='' :
+                    bookdata = Book.objects.filter(author=get_data['author'])
+                else: 
+                    bookdata = Book.objects.filter(genre=get_data['genre'],author=get_data['author'])
+        else:
+            if get_data['author']=='' :
+                if get_data['genre']=='' :
+                    bookdata = Book.objects.filter(title=get_data['title'])
+                else: 
+                    bookdata = Book.objects.filter(genre=get_data['genre'],title=get_data['title'])
+            else:
+                if get_data['genre']=='' :
+                    bookdata = Book.objects.filter(author=get_data['author'],title=get_data['title'])
+                else: 
+                    bookdata = Book.objects.filter(genre=get_data['genre'],author=get_data['author'],title=get_data['title'])
```

Comment:
You could have done this in a simpler way.
Also, you should have used `icontains` here instead of an exact match during search.